File: src/main/java/com/binarysushi/studio/webdav/clean/StudioCleanAction.java
package com.binarysushi.studio.webdav.clean;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.progress.PerformInBackgroundOption;
import com.intellij.openapi.progress.ProgressManager;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;


public class StudioCleanAction extends AnAction {

    public void actionPerformed(@NotNull AnActionEvent e) {
        Project project = e.getProject();

        ProgressManager.getInstance().run(
                new StudioCleanTask(project, "Cleaning cartridges...", true, PerformInBackgroundOption.ALWAYS_BACKGROUND)
        );
    }
}

File: src/com/binarysushi/studio/webdav/clean/StudioCleanAction.kt
package com.binarysushi.studio.webdav.clean

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.progress.PerformInBackgroundOption
import com.intellij.openapi.progress.ProgressManager

class StudioCleanAction : AnAction() {
    override fun actionPerformed(e: AnActionEvent) {
        val project = e.project
        ProgressManager.getInstance().run(
            StudioCleanTask(project, "Cleaning cartridges...", true, PerformInBackgroundOption.ALWAYS_BACKGROUND)
        )
    }
}
